### PR TITLE
Story/claudia 5735 filter package by os

### PR DESCRIPTION
**Pull request by hmunfru**
### REVIEWERS

@jesuspg  @flopezag 
### DESCRIPTION

Update murano package for including image OS as tag
### TESTING

Done locally

### CODE REVIEW

File: package-generator/model/product_package.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import shutil
 import errno
 import os
-from util import utils
+from util import utils_file as utils
 from model.cookbook import Cookbook
 
 PACKAGES_FOLDER = "./../murano-apps"
```
Comment (jesuspg) [inline package-generator/model/product_package.py:31]: remove old CONSTANT

Comment (hmunfru) [inline package-generator/model/product_package.py:31]: PACKAGE_FOLDERS is necessary 
 self.package_folder = (PACKAGES_FOLDER + "/" +
                               product.product_name + "/")
        self.package_classes = self.package_folder + "Classes/"
        self.package_classes_file = (self.package_classes +
                                     self.product.product_name + ".yaml")

Comment (flopezag) [inline package-generator/model/product_package.py:31]: Well, I have here two comments:
1) Why do you not define a TEMPLATE_FOLDER = 'template/PackageTemplate' and them PACKAGE_TEMPLATE_CLASS = TEMPLATE_FOLDER + '/Classes/GE_name.yaml' and so on.

2) I see several strings like '{GE_name}', '{GE_installator}' and so on why do you not defile a constant for them in a configuration file?

File: package-generator/tests/unit/test_cookbook.py [hunk 2]
```diff
@@ -184,16 +184,16 @@ def test_cookbook_puppet(self, mock_json, mock_open, mock_exists):
...
+        self.assertEquals(cookbook.url, COOKBOOK_URL)
+        self.assertEquals(len(cookbook.cookbook_childs), 2)
         i = 1
-        for cookbook_child in cookbook.get_cookbooks_child():
-            self.assertEquals(cookbook_child.get_cookbook_name(),
+        for cookbook_child in cookbook.cookbook_childs:
+            self.assertEquals(cookbook_child.name,
                               COOKBOOK_CHILD + str(i))
-            self.assertEquals(cookbook_child.get_url(),
+            self.assertEquals(cookbook_child.url,
```
Comment (flopezag) [inline package-generator/tests/unit/test_cookbook.py:194]: Why not:

'http://child%s.git' % str(i)

File: package-generator/tests/unit/test_product.py [hunk 3]
```diff
@@ -114,3 +124,13 @@ def test_ports(self):
         product = Product(PRODUCT_NAME, PRODUCT_VERSION, metadatas)
         self.assertEquals(len(product.get_tcp_ports()), 2)
         self.assertEquals(len(product.get_udp_ports()), 1)
+
+    @mock.patch('util.utils_clients.util_apis')
+    def test_obtain_images(self, mock_image):
+        """test the functionalities to get product images"""
+        mock_image._get_image_name.return_value = "nameID"
+        Config.Clients = mock_image
```
Comment (flopezag) [inline package-generator/tests/unit/test_product.py:132]: Maybe I do not see anything in the code But you are assiging the mock_image to the Config.Clients but you are using the Product class. I do not see the relationship between Config and Product.

File: package-generator/util/utils_file.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    """
+    return (url.startswith(("git://",
+                            "git+http://", "git+https:/"))
+            or url.endswith('.git'))
+
+
+def get_name_folder(url):
+    """
+    It returns the name of the folder for the url repository.
+    :param url:
```
Comment (flopezag) [inline package-generator/util/utils_file.py]: description of parameters...

File: package-generator/util/utils_file.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    end = splits[1]
+    if ".git" in end:
+        return end[0:end.find(".git")]
+    return end
+
+
+def read_metadata_puppet(url_file):
+    """
+    It reads the metadata.json from a git or svn repository .
+    :param url_file:
```
Comment (flopezag) [inline package-generator/util/utils_file.py]: description of parameters...

File: package-generator/util/utils_file.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    :return: the metadata in string
+    """
+    read_metadata(url_file, "metadata.json")
+
+
+def read_metadata_chef(url_file):
+    """
+    It reads the metadata.rb from a git or svn repository .
+    :param url_file:
```
Comment (flopezag) [inline package-generator/util/utils_file.py]: idem

File: package-generator/util/utils_file.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    :return: the metadata in string
+    """
+    read_metadata(url_file, "metadata.rb")
+
+
+def read_metadata(url_file, metadata_file):
+    """
+    It reads the metadata.rb from a git or svn repository .
+    :param url_file:
```
Comment (flopezag) [inline package-generator/util/utils_file.py]: idem + missing one

File: package-generator/util/utils_file.py [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    """
+    g = Github(user_github, password_github)
+    for repo in g.get_user().get_repos():
+        if repo.url == repo_url:
+            repo.create_pull("New update in Murano-packages",
+                             "Created by package-generator", branch, "develop")
+
+
+def download_files_git(url_file):
+    folder = COOKBOOK_FOLDER + get_name_folder(url_file)
```
Comment (flopezag) [inline package-generator/util/utils_file.py:145]: docstring

File: package-generator/util/utils_file.py [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    if is_git_repository(url_file):
+        if not os.path.exists(folder):
+            git.Git().clone(url_file, folder)
+
+
+def create_branch():
+    """
+    It creates a brach in the git repo and upload it
+    into github
+    :return:
```
Comment (flopezag) [inline package-generator/util/utils_file.py]: description

File: package-generator/util/utils_file.py [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+    """
+    repo = git.repo.Repo("./../")
```
Comment (flopezag) [inline package-generator/util/utils_file.py]: it is very dangerous to do in this way, due to it depends in which directory are you running the scripts.

File: package-generator/model/product_package.py (hunk 1 above)
Comment (hmunfru) [inline package-generator/model/product_package.py:31]: mm I don't like to define strings in a configuratino files. IMO configuration files are for values which can change, in this case, this value is not going to change. I agree in having a variable like: 
REPLACE_GE_NAME = {GE_name} or something like that. Take into account, these values should be in the Template files.

File: package-generator/tests/unit/test_cookbook.py (hunk 2 above)
Comment (hmunfru) [inline package-generator/tests/unit/test_cookbook.py:194]: yes. totally agreed

File: package-generator/tests/unit/test_product.py (hunk 3 above)
Comment (hmunfru) [inline package-generator/tests/unit/test_product.py:132]: Product needs to obtain the image name for glance. To do it, I am using some clients, whose configuration (passwodr, username,...) is in Config..

File: package-generator/util/utils_file.py (hunk 10 above)
Comment (hmunfru) [inline package-generator/util/utils_file.py]: OK. I will include a path parameter. But thinking on package-generator, you should consider that the git repo is in ./../

File: package-generator/util/utils_file.py [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+    if is_git_repository(url_file):
+        if not os.path.exists(folder):
+            git.Git().clone(url_file, folder)
+
+
+def create_branch(folder):
+    """
+    It creates a branch in the git repo and upload it
+    into github
+    :return:
```
Comment (flopezag) [inline package-generator/util/utils_file.py]: missing folder param description and return description.

File: package-generator/util/utils_file.py [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+    repo.commit()
+    repo.remotes.origin.push(new)
+    return str_branch
+
+
+def delete_branch(branch):
+    """
+    It deletes the branch in the git repo
+    :param branch: the branch to delete
+    """
```
Comment (flopezag) [inline package-generator/util/utils_file.py:182]: add return: Nothing.

File: package-generator/util/utils_file.py [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+        if repo.url == repo_url:
+            repo.create_pull("New update in Murano-packages",
+                             "Created by package-generator", branch, "develop")
+
+
+def download_files_git(url_file):
+    """
+    It download a file repository
+    :param url_file: the git repository url
+    """
```
Comment (flopezag) [inline package-generator/util/utils_file.py:144]: Please add return description in the doc of the method.

File: package-generator/model/product_package.py (hunk 1 above)
Comment (flopezag) [inline package-generator/model/product_package.py:31]: NTC

**Comment by jesuspg** [comment]
LGTM

**Pull request merged**